### PR TITLE
Fix silu(-inf) returning NaN instead of the limit 0

### PR DESCRIPTION
**Pull request by aws-alexk**
Fixes #160876

### Problem

`torch.nn.functional.silu(-inf)` returns `nan` instead of `0`.

ref: https://www.wolframalpha.com/input?i=lim+x+*+sigmoid%28x%29+as+x+-%3E+-inf

`silu(x) = x * sigmoid(x) = x / (1 + exp(-x))`. Mathematically `silu(x) -> 0` as `x -> -inf`. But evaluated directly at exactly `-inf`, the kernel computes `-inf / (1 + exp(inf)) = -inf / inf = nan`.

```python
>>> import torch
>>> torch.nn.functional.silu(torch.tensor(float("-inf")))
tensor(nan)   # expected: tensor(0.)
```

(Large finite-negative inputs already flush to `-0.0` correctly; only exactly `-inf` is affected.)

### Fix

Guard the `-inf` input so the kernels return the limit `0`:

- **CPU** (`aten/src/ATen/native/cpu/Activation.cpp`) — both the reduced-float (bf16/fp16) and floating/complex dispatch blocks, in the scalar and `Vectorized` lambdas. The vectorized path uses `blendv(y, 0, x == -inf)` to replace the affected lanes.
- **CUDA** (`aten/src/ATen/native/cuda/ActivationSiluKernel.cu`) — the `GPU_LAMBDA`.

Both guards are gated on `!c10::is_complex<scalar_t>` via `if constexpr`, so complex `silu` keeps its existing behavior (there is no `-inf` ordering for complex).

The `nn.functional.silu` OpInfo reference in `common_methods_invocations.py` is updated to match the limit (`np.where(x == -inf, 0.0, ...)`), so `test_reference_numerics_extremal` agrees with the op.

### Testing

Built CPU-only and ran locally:

- `python test/test_unary_ufuncs.py -k test_silu` — 4/4 pass (incl. a new `silu(-inf) == 0` regression assert added to `test_silu`).
- `python test/test_unary_ufuncs.py -k test_reference_numerics_extremal` (silu) — `float16`, `bfloat16`, `float32`, `float64`, `complex128` pass; `complex64` skip is pre-existing.

The CUDA kernel change mirrors the CPU logic; I don't have a GPU locally, so it relies on CI.

---
_Disclosure: this change was prepared with the assistance of an AI coding tool; I've reviewed and validated it._

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @aditew01

### CODE REVIEW

File: aten/src/ATen/native/cpu/Activation.cpp [hunk 1]
```diff
@@ -1133,16 +1133,28 @@ void silu_kernel(TensorIteratorBase& iter) {
...
+          [kOneVec, kZeroVec, kNegInfVec](
+              Vectorized<scalar_t> x_vec) -> Vectorized<scalar_t> {
             auto [x_vec0, x_vec1] = convert_to_float<scalar_t>(x_vec);
-            return convert_from_float<scalar_t>(
-              x_vec0 / (kOneVec + x_vec0.neg().exp()),
-              x_vec1 / (kOneVec + x_vec1.neg().exp()));
+            auto y0 = x_vec0 / (kOneVec + x_vec0.neg().exp());
+            auto y1 = x_vec1 / (kOneVec + x_vec1.neg().exp());
+            // Replace the -inf lanes (NaN above) with the limit, 0.
+            y0 = Vectorized<float>::blendv(y0, kZeroVec, x_vec0 == kNegInfVec);
```
Comment (Skylion007) [inline aten/src/ATen/native/cpu/Activation.cpp:1155]: Why vectorized<Float> over scalar_t? Also shouldn't it be if constexpr hasQuietNaN from std::numeric_limits?

Comment (aws-alexk) [reply aten/src/ATen/native/cpu/Activation.cpp:1155]: > Why vectorized over scalar_t?

in the reduced-float branch the input is already upcast to Vectorized<float> by `convert_to_float` and all the silu math runs in fp32, so the compare-and-blend has to be Vectorized<float> too (there's no half-precision scalar_t vector at that point)

> shouldn't it be if constexpr hasQuietNaN from std::numeric_limits?

added comment for this, `if constexpr (has_quiet_NaN` breaks the build.
c10::complex inherits std::numeric_limits from its value type, so has_quiet_NaN is true for complex and wouldn't exclude it; the complex branch would then instantiate Vectorized<c10::complex>::blendv, which doesn't compile

Comment (aws-alexk) [reply aten/src/ATen/native/cpu/Activation.cpp:1155]: any updates? @Skylion007 @eqy @syed-ahmed @Aidyn-A

Comment (aws-alexk) [reply aten/src/ATen/native/cpu/Activation.cpp:1155]: Bump on this @drisspg
